### PR TITLE
MueLu & MiniEM changes

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/muelu @trilinos/panzer 

## Description
- Fix an issue that made RHS assembly slow on Cuda for MiniEM.
- Use random current forcing term instead of Gaussian Pulse (which is mesh specific).
- Add timers to Thyra Belos adapter to track down why the CG remainder is exploding on Cuda.
- When running with stacked timer, don't create summary timers in MueLu.
- Do not rebalance on size 1 communicators, even if it's switched on.
- Skip one unneeded initialization in MueLu's iterate.
- Use cached vector in Ifpack2 MTGS.

### CODE REVIEW

File: packages/ifpack2/src/Ifpack2_Relaxation_def.hpp [hunk 1]
```diff
@@ -1916,8 +1916,8 @@ MTGaussSeidel (const Tpetra::MultiVector<scalar_type,local_ordinal_type,global_o
     }
   }
   else { // Column Map and domain Map are _not_ the same.
-    //X_colMap = crsMat->getColumnMapMultiVector (X);
-    X_colMap = rcp (new MV (colMap, X.getNumVectors ()));
+    updateCachedMultiVector(colMap,as<size_t>(X.getNumVectors()));
```
[the remark this answers is not in the document]
Comment (cgcgcg) [reply packages/ifpack2/src/Ifpack2_Relaxation_def.hpp]: This is the mechanism that was already implemented for regular GS and that we are simply reusing:
https://github.com/trilinos/Trilinos/blob/e1efc149b50f9215e5d1a2c53e82ae64dc879971/packages/ifpack2/src/Ifpack2_Relaxation_def.hpp#L153-L159

**Comment by cgcgcg** [comment]
@jhux2 Yep, it does.

**Comment by jhux2** [comment]
> Yep, it does.
:crying_cat_face:

File: packages/ifpack2/src/Ifpack2_Relaxation_def.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhoemmen) [reply packages/ifpack2/src/Ifpack2_Relaxation_def.hpp]: @cgcgcg Cool, that's fine then :-)

**Pull request merged**